coalib/settings/ConfigurationGathering.py

```diff
@@ -227,6 +227,13 @@ def get_config_directory(section):
     >>> get_config_directory(section) == section['files'].origin
     True
 
+    The user can manually set a project directory with the ``project_dir``
+    setting:
+
+    >>> section.append(Setting('project_dir', os.path.abspath('/tmp'), '/')
+    >>> get_config_directory(section) == os.path.abspath('/tmp')
+    True
+
```

Comment:
You should probably summarize the order of priority somewhere.

 1. `config_dir` from coafile
 2. origin of `file` from coafile
 3. coafile itself, if coafile is a directory
 4. coafile's directory if it is a file